### PR TITLE
1482: Add GitHub actions for header guards and license check

### LINKED ISSUE #1482: Github action to check for license correctness and header guards
**What Needs to be Done?**

We want to verify that the license matches the license-template file and that they header guard is unique and is composed of the directories and file name.

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixes #1482 

---
### Changes:
- Add GitHub Action to check License header in `src`, `tests/unit`, `tests/perf`, `tutorial` and `examples` subdirectories for all *.h and *.cc files
- Add GitHub Action to check header guards in `src`, `tests/unit` and  `tests/perf` directories. In order for this Action to work, I had to unify all header guards. 

Few examples of header guards for different files:

| File Path                 | Header Guards              |
| :--------------------------: | :--------------------: |
| `src/vt/vrt/collection/manager.h`            | `INCLUDED_VT_VRT_COLLECTION_MANAGER_H`     | 
| `tests/unit/collection/test_collection_common.h`            | `INCLUDED_UNIT_COLLECTION_TEST_COLLECTION_COMMON_H`     | 
| `tests/perf/common/test_harness.h`            | `INCLUDED_PERF_COMMON_TEST_HARNESS_H`     | 

- Remove some dead files that were empty (only included the license)
image

### CODE REVIEW

File: .github/workflows/check-include-guards.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          for sub_dir in "src" "tests/unit" "tests/perf"
+          do
+            wrong_files+=$(checkguard -r "$sub_dir" -p "path -1 | prepend INCLUDED_ | upper")
+          done
+
+          if [[ $wrong_files ]]; then
+            echo "Files with wrong header guard:"
+            printf "%s\n" "$wrong_files"
+            exit 1
+          fi
```
Comment (jstrzebonski) [inline .github/workflows/check-include-guards.yml:33]: There's a `scripts\` directory. You might want to move this code and license code there..

**Comment by jstrzebonski** [comment]
Since with now have this great automagical tool, can we (should we?) change copyright year in the template to `2019-2021`?

File: .github/workflows/check-include-guards.yml (hunk 1 above)
Comment (JacobDomagala) [reply .github/workflows/check-include-guards.yml]: Done

**Review by lifflander — APPROVED** [review]
This looks great to me! Is the `checkguard` program the thing that automatically makes sure the inclusion guards are correct? I couldn't tell how it enforces a certain inclusion name/style on each file?

**Comment by JacobDomagala** [comment]
> This looks great to me! Is the `checkguard` program the thing that automatically makes sure the inclusion guards are correct? I couldn't tell how it enforces a certain inclusion name/style on each file?

It is using [its own pattern](https://github.com/cgmb/guardonce/blob/master/docs/PatternLanguage.md) to check whether the guards are valid. 
I'm using the following `path -1 | prepend INCLUDED_ | upper` which translates to:
| File Path                 | Header Guards              |
| :--------------------------: | :--------------------: |
| `src/vt/vrt/collection/manager.h`            | `INCLUDED_VT_VRT_COLLECTION_MANAGER_H`     | 
| `tests/unit/collection/test_collection_common.h`            | `INCLUDED_UNIT_COLLECTION_TEST_COLLECTION_COMMON_H`     | 
| `tests/perf/common/test_harness.h`            | `INCLUDED_PERF_COMMON_TEST_HARNESS_H`     |

**Comment by JacobDomagala** [comment]
Updated the copyright year to `2019-2021` in all files that include the vt's license (including our main LICENSE file)

**Pull request merged**